test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -5129,9 +5199,88 @@ def test_manage_rlimit_as(self):
         job_out = '\n'.join(result['out'])
         self.logger.info("job_out=%s" % job_out)
         # ulimit reports kb, not bytes
-        self.assertTrue(str(300 * 1024) in job_out)
-        self.logger.info("Job that requests 300mb pvmem "
-                         "correctly has 300mb RLIMIT_AS")
+        self.assertTrue(str(400 * 1024) in job_out)
+        self.logger.info("Job that requests 400mb pvmem "
+                         "correctly has 400mb RLIMIT_AS")
+
+    def test_cgroup_mount_paths(self):
+        """
+        Test to see if the cgroup hook picks the shortest path,
+        but also if it can be overrided in the config file
+        """
+
+        if self.du.isdir(self.hosts_list[0], '/dev/tstc'):
+            self.skipTest('Test requires /dev/tstc not to exist')
+        if self.du.isdir(self.hosts_list[0], '/dev/tstm'):
+            self.skipTest('Test requires /dev/tstm not to exist')
+
+        self.load_config(self.cfg17)
+
+        self.du.mkdir(hostname=self.hosts_list[0],
+                      path='/dev/tstm', mode=0o0755, sudo=True)
+        self.du.run_cmd(self.hosts_list[0],
+                        ['mount', '-t', 'cgroup', '-o',
+                         'rw,nosuid,nodev,noexec,relatime,seclabel,memory',
+                         'cgroup', '/dev/tstm'],
+                        sudo=True)
+
+        self.du.mkdir(hostname=self.hosts_list[0],
+                      path='/dev/tstc', mode=0o0755, sudo=True)
+        self.du.run_cmd(self.hosts_list[0],
+                        ['mount', '-t', 'cgroup', '-o',
+                         'rw,nosuid,nodev,noexec,relatime,seclabel,cpuset',
+                         'cgroup', '/dev/tstc'],
+                        sudo=True)
+
+        # sleep 2s: make sure no old log lines will match 'begin' time
+        time.sleep(2)
+        begin = int(time.time())
+        # sleep 2s to allow for small time differences and rounding errors
+        time.sleep(2)
+
+        a = {'Resource_List.select':
+             "1:ncpus=1:host=%s" % self.hosts_list[0]}
+        j = Job(TEST_USER, attrs=a)
+        j.create_script(self.sleep100_job)
+        jid = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid)
+        failure = False
+
+        try:
+            self.moms_list[0].log_match(msg='create_job: Creating directory '
+                                            '/sys/fs/cgroup/cpuset/'
+                                            'pbs_jobs.service/jobid/%s'
+                                            % jid,
+                                        n='ALL', starttime=begin,
+                                        max_attempts=1)
```

Comment:
Are we guaranteed that the message will be printed to the log by the time the job is in the R state?  Or should we increase max_attempts to more than 1 for slow test machines?  Same comment for line 5265.

Reply:
Yes, we are guaranteed the message will be printed.